libfreerdp/core/certificate.c

```diff
@@ -351,19 +430,18 @@ static void certificate_free_x509_certificate_chain(rdpX509CertChain* x509_cert_
 	{
 		for (UINT32 i = 0; i < x509_cert_chain->count; i++)
 		{
-			struct rdp_CertBlob* element = &x509_cert_chain->array[i];
-			free(element->data);
+			rdpCertBlob* element = &x509_cert_chain->array[i];
+			cert_blob_free(element);
 		}
 	}
 
 	free(x509_cert_chain->array);
-	free(x509_cert_chain);
 }
 
 static BOOL certificate_process_server_public_key(rdpCertificate* certificate, wStream* s,
                                                   UINT32 length)
 {
-	BYTE magic[4] = { 0 };
+	char magic[sizeof(rsa_magic)] = { 0 };
 	UINT32 keylen = 0;
```

Comment:
perhaps having some define instead of `sizeof(rsa_magic)` would make the code nicer

Reply:
prefer the `sizeof()` where possible, as it does not introduce a define which could get out of sync